astropy/io/fits/header.py

```diff
@@ -1694,11 +1694,11 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
 
         keyword = self._cards[idx].keyword
         keyword = Card.normalize_keyword(keyword)
-        repeat = self._keyword_indices[keyword].index(idx)
+        repeat = len(self._keyword_indices[keyword]) - 1
```

Comment:
header._keyword_indices[keyword] gives a list of the indices where a particular keyword is present. Now it did not give the length as explained in #5827 . 

```>>> hdr = pf.Header([ ('MOONFRAC', 0.604829), ('MOONDIST', 82.5171118)])
>>> hdr.append()
>>> hdr.append()
>>> hdr._keyword_indices['']
[2,3]
```
So number of times a header is repeated is just `len(hdr._keyword_indices[''])-1` which in this case for '' will be 1 and for others will be 0.